### DISCUSSION TITLE
FastLogin don't work on 1.19 NAT server -- Invaild session

### DISCUSSION THREAD

**Discussion by hank9999** [games647/FastLogin#844] [Q&A] [answered]
Hi! I have tried latest dev ProtocolLib and FastLogin, when i try joining server, it show Invalid session. My client ip in log is right.
but, when i try it in my local server, it works. My remote server is a nat server.
![111111111111](https://user-images.githubusercontent.com/20784591/178468293-3b02c567-2e30-4be1-a6d2-6918a2d19b32.png)

**Comment by TuxCoding** [comment, marked as answer]
You need to forward the original IP to the Minecraft server. Otherwise features like IP bans, session based logins like in AuthMe will break or even be security risk. This includes FastLogin. Here in this case Mojang rejects your connection, because the IP address incoming to the server is different to the IP address the client made connecting to Mojang. Therefore suspected MITM proxy is assumed.

If you really want skip this check, you can disable the proxy check in the configuration file.

**Reply by hank9999** [reply]
Thanks for your answer. it works when checking disabled. But the original ip seems to be forward to the server, ip in logs are original.

**Reply by TuxCoding** [reply]
Is the IP address that FastLogin used a local one? The message clearly indicates that the connection was rejected by Mojang. Maybe there is an encoding issue of FastLogin during the transmission of it.

**Reply by TuxCoding** [reply]
How is the original IP address forwarded to the server. Maybe FastLogin didn't received that notification.

**Reply by hank9999** [reply]
> How is the original IP address forwarded to the server. Maybe FastLogin didn't received that notification.

IP address showed by FastLogin's log is original. I do not know how my idc do, but may use iptables etc.
